internal/config/config_test.go

```diff
@@ -203,67 +532,214 @@ func TestRead(t *testing.T) {
+				args: args{
+					path: path,
+					cfg:  cfg,
+				},
+				beforeFunc: func(t *testing.T, _ args) {
+					t.Helper()
+					f, err := os.Create(path)
+					if err != nil {
+						t.Fatal(err)
+					}
+					defer f.Close()
+
+					f.WriteString(data)
+				},
+				afterFunc: func(t *testing.T, _ args) {
+					t.Helper()
+					if err := os.Remove(path); err != nil {
+						t.Fatal(err)
+					}
+				},
+				want: want{
+					want: &GlobalConfig{
+						Version: "v1.0.0",
+						TZ:      "UTC",
+						Logging: &Logging{
+							Logger: "glg",
+							Level:  "warn",
+							Format: "json",
+						},
+					},
+					err: nil,
+				},
+			}
+		}(),
+		func() test {
+			path := "read_test_config.yaml"
+			cfg := new(GlobalConfig)
+			return test{
+				name: "return no entry error when the file open fails",
+				args: args{
+					path: path,
+					cfg:  cfg,
+				},
+				want: want{
+					want: cfg,
+					err: &fs.PathError{
+						Op:   "open",
+						Path: "read_test_config.yaml",
+						Err:  syscall.ENOENT,
+					},
+				},
+			}
+		}(),
+		func() test {
+			path := "read_test_config.yaml"
+			data := "timezone\n:"
+			cfg := new(GlobalConfig)
+			return test{
+				name: "return yaml decode error when the contents of yaml is invalid",
+				args: args{
+					path: path,
+					cfg:  cfg,
+				},
+				beforeFunc: func(t *testing.T, _ args) {
+					t.Helper()
+					f, err := os.Create(path)
+					if err != nil {
+						t.Fatal(err)
+					}
+					defer f.Close()
 
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           args: args {
-		           path: "",
-		           cfg: nil,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+					f.WriteString(data)
+				},
+				afterFunc: func(t *testing.T, _ args) {
+					t.Helper()
+					if err := os.Remove(path); err != nil {
+						t.Fatal(err)
+					}
+				},
+				want: want{
+					want: cfg,
+					err:  errors.New("yaml: unmarshal errors:\n  line 1: cannot unmarshal !!str `timezone` into config.GlobalConfig"),
+				},
+			}
+		}(),
+		func() test {
+			path := "read_test_config.json"
+			data := "timezone\n:"
+			cfg := new(GlobalConfig)
+			return test{
+				name: "return json decode error when the contents of json file is invalid",
+				args: args{
+					path: path,
+					cfg:  cfg,
+				},
+				beforeFunc: func(t *testing.T, _ args) {
+					t.Helper()
+					f, err := os.Create(path)
+					if err != nil {
+						t.Fatal(err)
+					}
+					defer f.Close()
+
+					f.WriteString(data)
+				},
+				afterFunc: func(t *testing.T, _ args) {
+					t.Helper()
+					if err := os.Remove(path); err != nil {
+						t.Fatal(err)
+					}
+				},
+				want: want{
+					want: cfg,
+					err:  errors.New("skipThreeBytes: expect rue, error found in #2 byte of ...|timezone\n:|..., bigger context ...|timezone\n:|..."),
+				},
+			}
+		}(),
```

Comment:
I think you have to consider more input variables when the function wants to variable as `interface{}`.
`struct`, `slice`, `map`, `byte`, `string`, or etc.
how do you think? cc: @kevindiu

Reply:
I agree with you, to increase test coverage (not code coverage)

Reply:
thank you for your comment :bow:
I agree with yours. so I will fix it soon